### PR TITLE
interface\patient_file\summary library refactor

### PR DESCRIPTION
**Pull request by MigDinny**
#915 

Made refactors -> now using call_required_libraries

jquery upgraded to latest version

### CODE REVIEW

**Comment by teryhill** [comment]
@migdinny look at issue 416 This is where art and I did a lot of work on the jquery issue. There are some functions that need changing / updated to go to 3-1-1. This code is very old and changes , for a little while need to be done slowly.

**Comment by MigDinny** [comment]
@teryhill yes

I opened a patient and browsed through it with my console opened. No errors in the scope of this PR were found (some errors were found but they were found even when I changed my branch to master).

Please do not take my testings as 100% guaranteed because I'm new to the codebase and some files are probably out my 'sight'. It's hard for me to test a program I don't fully understand... I did my best and I will fix any bug that may occur anyway.

**Comment by aethelwulffe** [comment]
@robbyoconnor Did you run tests on this branch when you added the "tested"?  Because if you did, pretty pretty please also do a review!
  We need code reviews *seriously bad*.

File: interface/patient_file/summary/add_edit_amendments.php [hunk 1]
```diff
@@ -122,18 +123,12 @@
...
-<!-- supporting javascript code -->
-<script type="text/javascript" src="<?php echo $GLOBALS['webroot'] ?>/library/js/jquery-1.7.2.min.js"></script>
-<script type="text/javascript" src="<?php echo $GLOBALS['webroot'] ?>/library/textformat.js"></script>
-<script type="text/javascript" src="<?php echo $GLOBALS['webroot'] ?>/library/dialog.js"></script>
-
-<script type="text/javascript" src="<?php echo $GLOBALS['webroot'] ?>/library/js/jquery.datetimepicker.full.min.js"></script>
-
+<?php
+    html_header_show();
+    call_required_libraries(array('jquery-min-3-1-1', 'datepicker'));
```
Comment (aethelwulffe) [inline interface/patient_file/summary/add_edit_amendments.php]: Please add comment "requires textformat.js and dialog.js" or something to indicate that we need those unspecified libraries.

**Comment by MigDinny** [comment]
@aethelwulffe He didn't, I've asked him to add the label on my behalf. I have opened the files directly on the link and I've checked if all libraries were being called and I've checked the console. I don't know if this was the kind of testing needed, but it's hard for me to test using the software because I don't know what is the feature linked to a file...

Remove the labels if you think it's better.

File: interface/patient_file/summary/add_edit_amendments.php (hunk 1 above)
Comment (MigDinny) [reply interface/patient_file/summary/add_edit_amendments.php]: I can do it, but is it needed for the whole refactor? I mean, this function ALWAYS requires those 2 scripts... is this the only place that needs that comment?

**Comment by aethelwulffe** [comment]
I understand.
  We just have to do these js refactor PR's in order, and carefully.  We will get through them, but it will take a bit of time.

**Comment by MigDinny** [comment]
@aethelwulffe Yeah, I'm aware of that. We don't have to rush this.

By the way, can you take a look at my current task? I would like to have it approved so I could move on. One of the pr's there is probably going to be approved after GCI ...

**Comment by aethelwulffe** [comment]
Done.

**Comment by MigDinny** [comment]
@aethelwulffe i'll ship you a chocolate

**Comment by MigDinny** [comment]
solved merge conflicts

review the new commit (only made changes in headers.inc.php and demographics.php but i messed up here with git and i don't know if I did something wrong)

**Comment by MigDinny** [comment]
Solved conflicts but I'm requesting some tests on this because some conflicts were a bit weird to solve and I'm not sure if this is okay to be merged. stats_full.php is one of the files along with procedure_order/new.php i'm requesting

**Comment by aethelwulffe** [comment]
This PR was a bit of an overreach.  Most or all of this has been done elsewhere, and other stuff besides.  Just a little too big for folks to test all at once.

**Pull request closed without merging**